engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/mutable/SortNode.java

```diff
@@ -34,15 +42,22 @@ static SortNode build() {
   @Specialization
   Object doSortFunction(State state, Array self, Function comparator) {
     EnsoContext context = EnsoContext.get(this);
-    Comparator<Object> compare = getComparator(comparator, context, state);
-    return runSort(compare, self, context);
+    Comparator<Object> compare = getComparator(comparator, context, state, false);
+    return runSort(compare, self.getItems(), context);
   }
 
   @Specialization
   Object doAtomThis(State state, Atom self, Object that) {
     return EnsoContext.get(this).getBuiltins().nothing();
   }
 
+  @Specialization
+  Object doArrayOverBuffer(State state, ArrayOverBuffer self, Function comparator) {
```

Comment:
Can we sort JavaScript array? I'd expect specialization to `iop.hasArrayElements(self)`, but I don't see it anywhere.

Reply:
Now we can (see next specialization). However I'm now wondering if it is worth keeping this one since the next one would also handle `ArrayOverBuffer`. Except it would return sorted `Array` rather than `ArrayOverBuffer`, so it really depends how important that is.

Reply:
Exactly. `ArrayOverBuffer` is just an `iop.hasArrayElements(...)` and as such it could be handled by the next specialization in a generic way.